From f98e9dbdec48120e5841e45e2569eb8ccfa0d7b2 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:26:51 -0400
Subject: [net] genetlink: make genl_ops flags a u8 and move to end

Message-id: <502e8ba17643bfb439e0c141e4a17a4ef42056a0.1410528958.git.jbenc@redhat.com>
Patchwork-id: 93956
O-Subject: [RHEL7.1 net PATCH 09/23] genetlink: make genl_ops flags a u8 and move to end
Bugzilla: 1140656
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1140656
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit 3f5ccd06aecd0eb1651dd451439d5cb365170854
Author: Johannes Berg <johannes.berg@intel.com>
Date:   Thu Nov 14 17:14:47 2013 +0100

    genetlink: make genl_ops flags a u8 and move to end

    To save some space in the struct on 32-bit systems,
    make the flags a u8 (only 4 bits are used) and also
    move them to the end of the struct.

    This has no impact on 64-bit systems as alignment of
    the struct in an array uses up the space anyway.

    Signed-off-by: Johannes Berg <johannes.berg@intel.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/net/genetlink.h b/include/net/genetlink.h
index 6325f76..d7c88fa 100644
--- a/include/net/genetlink.h
+++ b/include/net/genetlink.h
@@ -121,15 +121,15 @@ static inline void genl_info_net_set(struct genl_info *info, struct net *net)
  * @ops_list: operations list
  */
 struct genl_ops {
- u8   cmd;
- u8   internal_flags;
- unsigned int  flags;
  const struct nla_policy *policy;
  int         (*doit)(struct sk_buff *skb,
            struct genl_info *info);
  int         (*dumpit)(struct sk_buff *skb,
       struct netlink_callback *cb);
  int         (*done)(struct netlink_callback *cb);
+ u8   cmd;
+ u8   internal_flags;
+ u8   flags;
 };
 
 int __genl_register_family(struct genl_family *family);
-- 
1.7.1